autogen/logger/file_logger.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import annotations
+
+import json
+import logging
+import os
+import threading
+import uuid
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple, Union
+
+from openai import AzureOpenAI, OpenAI
+from openai.types.chat import ChatCompletion
+
+from autogen.logger.base_logger import BaseLogger
+from autogen.logger.logger_utils import get_current_ts, to_dict
+
+from .base_logger import LLMConfig
+
+if TYPE_CHECKING:
+    from autogen import Agent, ConversableAgent, OpenAIWrapper
+
+logger = logging.getLogger(__name__)
+lock = threading.Lock()
+
+
+class FileLogger(BaseLogger):
+    def __init__(self, config: Dict[str, Any]):
+        self.config = config
+        self.session_id = str(uuid.uuid4())
+        self.log_file = self.config.get("filename", "runtime.log")
+
+    def start(self) -> str:
+        try:
+            with open(self.log_file, "a"):
+                pass
+        except Exception as e:
+            logger.error(f"[file_logger] Failed to create logging file: {e}")
+            raise e
+        finally:
+            return self.session_id
+
+    def log_chat_completion(
+        self,
+        invocation_id: uuid.UUID,
+        client_id: int,
+        wrapper_id: int,
+        request: Dict[str, Union[float, str, List[Dict[str, str]]]],
+        response: Union[str, ChatCompletion],
+        is_cached: int,
+        cost: float,
+        start_time: str,
+    ) -> None:
+        with lock:
```

Comment:
Opening, writing and closing the file on every log is going to be quite inefficient. You could consider using a built in python logger implementation here to reuse some of the buffering logic of it

Reply:
sure @jackgerrits , Thanks.